### ISSUE TITLE
bug(k8s): network policy query treats matchLabels entries independently

### ISSUE THREAD

**Issue by lokesh0186** [Checkmarx/kics#8107]
### Expected Behavior

The `Network Policy Is Not Targeting Any Pod` query should report a NetworkPolicy when no single Pod satisfies all entries in `spec.podSelector.matchLabels`.

Kubernetes label selectors combine the requirements with AND semantics. A Pod must match both `app: api` and `tier: backend` in the example below.

### Actual Behavior

KICS reports no result when one Pod matches `app: api` and a different Pod matches `tier: backend`, even though neither Pod matches the complete selector.

### Steps to Reproduce the Problem

Save this as `resources.yaml`:

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: api-frontend
  namespace: demo
  labels:
    app: api
    tier: frontend
spec:
  containers:
    - name: app
      image: example.invalid/app:1
---
apiVersion: v1
kind: Pod
metadata:
  name: web-backend
  namespace: demo
  labels:
    app: web
    tier: backend
spec:
  containers:
    - name: app
      image: example.invalid/app:1
---
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: api-backend
  namespace: demo
spec:
  podSelector:
    matchLabels:
      app: api
      tier: backend
  policyTypes:
    - Ingress
  ingress: []
```

Run only the affected query:

```console
docker run --rm -v "$PWD:/path" \
  checkmarx/kics@sha256:3e5a268eb8adda2e5a483c9359ddfc4cd520ab856a7076dc0b1d8784a37e2602 \
  scan -p /path/resources.yaml \
  --include-queries 85ab1c5b-014e-4352-b5f8-d7dea3bb4fd3 \
  --no-progress --no-color --minimal-ui
```

The result is:

```text
LOW: 0
TOTAL: 0
```

I also checked two controls. A Pod labeled with both `app: api` and `tier: backend` produces no result, as expected. A selector whose labels are not present on any Pod does produce results, confirming that the query is running.

The current query appears to test each `matchLabels` entry against the Pod inventory separately. That allows different Pods to satisfy different parts of the selector.

As an independent cross-check, public [IaC-Guard-V 0.1.0b1](https://github.com/lokesh0186/iac-guard-v/releases/tag/v0.1.0-beta.1) returns `VIOLATED` for both Pods with `IACGV_K8S_WORKLOAD_POLICY_SELECTED_V1`. Each witness shows one label matching and the other failing, so neither Pod is selected by the policy.

If this behavior is unintended, I can prepare a focused query regression and fix after the issue is reviewed.

### Specifications

- Version: official `checkmarx/kics:latest` image, KICS v2.1.20, image digest `sha256:3e5a268eb8adda2e5a483c9359ddfc4cd520ab856a7076dc0b1d8784a37e2602`
- Platform: Kubernetes
- Subsystem: query `85ab1c5b-014e-4352-b5f8-d7dea3bb4fd3`
- Current main checked: `fa915dda04498c3cf035caf8eac3abc5acb346cd`
- The query file in the image and current main has the same SHA256: `b746feaac4472d2f82a7fb774b8e12e626e4cf5e2bf8b8f315ffc12ef7597d2e`
